pages/users/[id]/index.js

```diff
@@ -0,0 +1,148 @@
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import styled from "styled-components";
+import { useSession } from "next-auth/react";
+import { StyledLink } from "@/components/StyledLink";
+import { StyledButton } from "@/components/StyledButton";
+import Image from "next/image";
+
+export default function User() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: user, isLoading, error } = useSWR(`/api/users/${id}`);
+  const { data: session } = useSession();
+  let showEditButton = false;
+
+  if (!isReady || isLoading || error ) return <h2>Loading...</h2>;
+
+  if ( session ) {
+    const isAdmin = session?.user.role === "admin";
+    const isCurrentUser = session?.user.name === user.name;
+    showEditButton = isAdmin || isCurrentUser;
+  }
+
+  async function deleteUser() {
+    try {
+      const response = await fetch(`/api/users/${id}`, {
+        method: "DELETE",
+      });
+      !response.ok ? console.error(response.error) : router.push("/")
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  return (
+    <StyledWrapper>
+      <h1>User Profile</h1>
+      <h2>Hello {user.name}</h2>
+      <StyledTable>
```

Comment:
I would consider using the `table` tag here. (Mostly for accessibility reasons). And then the subsequent sub-tags like `thead`, `tr`, `th` etc.

[table tag on mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table)

I am not saying this needs to be changed for this pull request, but definitely worth thinking about.

Reply:
it was a conscious decision to use no table tag, for use it responsive with more flexibility its more powerfull to use, dynamic elements like div span or what else as an static table element. So if you need a look like a table, you can its modify in css like flex, grid or even so on display table. So you can use the power of both worlds